---
title: "Material and Methods"
author: "Florent Chuffart"
date: "`r Sys.Date()`"
output:
  rmarkdown::html_document:
    toc: true
    toc_float: true
    toc_depth: 3
    number_sections: true
---


```{r, echo=FALSE, eval=TRUE}
knitr::opts_chunk$set(collapse=TRUE, comment = "#>", fig.width=9, fig.height=6, eval=TRUE, echo=FALSE, results="hide")
source("config")
```



# Material and Methods



## Sequencing

**FOLLOWING CONCERNS TGML PLATFORM** Total RNA was purified by trizol extraction followed by Qiagen RNeasy mini kit protocol. 
Removal of most rRNA was proceeded using NEBNext rRNA Depletion Kit Ribodepletion. 
Quality was verified by Agilent Bioanalyzer 2100. 
Libraries were prepared using NEBNext Ultra Directional RNA Library Prep kit for Illumina.
Base calling was performed using RTA version 2.*

Total RNA was purified by XXX.
Removal of most rRNA was proceeded using XXX.
Quality was verified by XXX. 
Libraries were prepared using XXX.
Base calling was performed XXX.

Stranded paired-end 2x75bp reads were produced by Illumina NextSeq500. 

**Alignment**

The sequenced reads from the raw sequence (.fastq files) were aligned on the `r annotation` `r version` genome using `STAR` (2.7.1a) [PMID:23104886] to produce bam files.

**Counts**

The aligned reads (.bam files) were counted using `HTSeq` framework (0.11.2) [PMID:25260700], with options:** -t exon -f bam -r pos --stranded=`r strand` -m intersection-strict --nonunique none


**Normalization and differential analysis**

The normalization, pseudo-log transformation of the read counts, and the differential analysis were performed using the R software [R Core Team. R: A language and environment for statistical computing. Vienna, Austria : R Foundation for Statistical Computing, 2017], `DESeq2` (1.22.2) [PMID:20979621, PMID:25516281] and `SARTools` [PMID:27280887] packages.


**Grouping genes/features according to their quartile of expression**

The features corresponding to non-coding transcripts and/or with zero counts were filtered out, in order to retain all non-zero counts protein-coding genes from NCBI Reference Sequence Database (RefSeq). The corresponding genes and features were ranked according to their DESeq2 normalized expression mean value in the samples corresponding to the 4 replicates of liver from wild-type mice under normal diet, and allocated into four subsets corresponding to the quartiles of this expression and exported as .bed files.

[abs_expr_q1_0.bed](./abs_expr_q1_0.bed) 
[abs_expr_q2_0.bed](./abs_expr_q2_0.bed)
[abs_expr_q3_0.bed](./abs_expr_q3_0.bed)
[abs_expr_q4_0.bed](./abs_expr_q4_0.bed)






# Session Information

```{r results="verbatim"}
sessionInfo()
date()
```



























